sympy/ntheory/qs.py

```diff
@@ -0,0 +1,234 @@
+from sympy import isprime, log, sqrt, exp, sieve, gcd, mod_inverse, ceiling
+from sympy.ntheory.residue_ntheory import _sqrt_mod_prime_power
+
+class Polynomial:
```

Comment:
This name looks too general and may be confusing. We only have quadratic polynomials with three terms (trinomials).

Reply:
i will change that